File: Frontend/types/doctor.ts
export enum UrgentCaseMode {
  UrgentNext = 0,
  UrgentBucket = 1,
  UrgentFront = 2,
}

export enum DoctorCompensationMode {
  Salary = 0,
  Percentage = 1,
  FixedPerVisit = 2,
}

export interface IDoctorCompensationHistoryItem {
  id: string
  mode: DoctorCompensationMode
  value: number
  effectiveFrom: string
  changedByUserId: string
  notes?: string | null
  createdAt: string
}

export interface IDoctorService {
  id?: string
  serviceName: string
  price: number
  durationMinutes?: number
  isActive: boolean
}

export interface IDoctorVisitConfig {
  bloodPressure: boolean
  heartRate: boolean
  temperature: boolean
  weight: boolean
  height: boolean
  bmi: boolean
  bloodSugar: boolean
  oxygenSaturation: boolean
  respiratoryRate: boolean
}

export interface IDoctor {
  id: string
  userId: string
  name: string
  username: string
  specialty: string
  phone: string
  photoUrl: string | null
  isEnabled: boolean
  bio?: string
  urgentCaseMode: UrgentCaseMode
  urgentInsertAfterCount: number
  supportsUrgent: boolean // <-- جديد
  avgVisitDurationMinutes: number
  compensationMode: DoctorCompensationMode
  compensationValue: number
  compensationEffectiveFrom: string
  services: IDoctorService[]
  compensationHistory?: IDoctorCompensationHistoryItem[]
  visitFieldConfig: IDoctorVisitConfig
  createdAt: string
}

File: src/EliteClinic.Domain/Entities/Doctor.cs
using EliteClinic.Domain.Enums;

namespace EliteClinic.Domain.Entities;

public class Doctor : TenantBaseEntity
{
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Specialty { get; set; }
    public string? Phone { get; set; }
    public string? Bio { get; set; }
    public string? PhotoUrl { get; set; }
    public bool IsEnabled { get; set; }
    public UrgentCaseMode UrgentCaseMode { get; set; }
    public bool UrgentEnabled { get; set; }
    public int UrgentInsertAfterCount { get; set; }
    public int AvgVisitDurationMinutes { get; set; }
    public DoctorCompensationMode CompensationMode { get; set; }
    public decimal CompensationValue { get; set; }
    public DateTime CompensationEffectiveFrom { get; set; }

    // Navigation
    public ApplicationUser User { get; set; } = null!;
    public ICollection<DoctorService> Services { get; set; } = new List<DoctorService>();
    public ICollection<DoctorCompensationHistory> CompensationHistory { get; set; } = new List<DoctorCompensationHistory>();
    public ICollection<DoctorBranchSchedule> BranchSchedules { get; set; } = new List<DoctorBranchSchedule>();
    public DoctorVisitFieldConfig? VisitFieldConfig { get; set; }

    public Doctor()
    {
        IsEnabled = true;
        UrgentCaseMode = UrgentCaseMode.UrgentNext;
        UrgentEnabled = true;
        UrgentInsertAfterCount = 0;
        AvgVisitDurationMinutes = 15;
        CompensationMode = DoctorCompensationMode.Percentage;
        CompensationValue = 0;
        CompensationEffectiveFrom = DateTime.UtcNow.Date;
    }
}
